Automattic/woocommerce-payments
# Replace international billing address filter with IP and shipping country mismatch filter

Fixes Automattic/woocommerce-payments-server#3194

#### Changes proposed in this Pull Request

This PR replaces the international billing address filter with the IP country filter described in pcqRLn-1wY-p2#comment-2059. It should basically check if the IP country is the same with the billing country and run relative actions on the transaction.

It should be tested in conjunction with Automattic/woocommerce-payments-server#3197 to be able to run the filter.









#### Testing instructions





* Checkout this branch on your client 
* Checkout the branch in Automattic/woocommerce-payments-server#3197 on your server.
* Start listening on webhooks with `npm run listen` on the server.
* Change the version string in `woocommerce-payments.php` file to 5.7.0 (on your client). 
* Enable fraud protection meta on your client.
* Go to advanced fraud protection settings page, and check if the new filter displays as described in the thread linked in the PR description.
* Disable all other filters, just enable IP country filter.
* Leave block checkbox unchecked. 
* Go to your store, add an item to the cart, and go to checkout. 
* On the checkout page, select a country that's different from your IP country and complete the checkout.
* You should get an on-hold order.
* Go to it's transaction details, see if the outcome is correctly spelled there.

Repeat this test with blocking active. 

Also you should repeat both block and review tests with: 
- Legacy UPE enabled
- Legacy UPE disabled
- Legacy UPE enabled with a LPM, should checkout but not block & put to review
- Split UPE enabled with a LPM, should work like the first two

-------------------


**Post merge**
